public/app/plugins/panel/piechart/module.tsx

```diff
@@ -35,19 +42,55 @@ export const plugin = new PanelPlugin<PieChartOptions>(PieChartPanel)
         },
         defaultValue: PieChartType.Pie,
       })
-      .addBooleanSwitch({
-        name: 'Show name',
-        path: 'labelOptions.showName',
-        defaultValue: true,
+      .addCustomEditor<SelectFieldConfigSettings<PieChartLabels>, PieChartLabels[]>({
+        name: 'Labels',
+        path: 'displayLabels',
+        id: 'displayLabels',
+        defaultValue: [],
+        editor: standardEditorsRegistry.get('multi-select').editor,
+        description: 'Select the labels to be displayed in the pie chart',
+        settings: {
+          options: [
+            { value: PieChartLabels.Percent, label: 'Percent' },
+            { value: PieChartLabels.Name, label: 'Name' },
+            { value: PieChartLabels.Value, label: 'Value' },
+          ],
+        },
```

Comment:
Since the new std editor is added, maybe intead of adding custom editor here you could extend the Options API with `addMultiSelect` method?

Also- if you could open the options API changes as a separate PR it would be super helpful, as it feels like an issue on its own:)

Reply:
Made another draft PR for it here: #31501
Will add an addMultiSelect method there